sui_core/src/authority.rs

```diff
@@ -355,6 +359,7 @@ impl AuthorityState {
             transaction_digest,
         );
         let effects = execution_engine::execute_transaction_to_effects(
+            shared_object,
```

Comment:
So we explicitly pass this argument to avoid re-calling `.compute_object_reference()` within the executing engine?

Reply:
Yeah basically -- computing digests is not for free. So this leads to computing once, and allocating once.

Reply:
cool just making sure I didn't miss anything important